cmd/testrunner.go

```diff
@@ -88,11 +89,32 @@ func setupTestCommand() *cobraext.Command {
 			Args:  cobra.NoArgs,
 			RunE:  action,
 		}
-
 		if runner.CanRunPerDataStream() {
 			testTypeCmd.Flags().StringSliceP(cobraext.DataStreamsFlagName, "d", nil, cobraext.DataStreamsFlagDescription)
 		}
 
+		if runner.CanRunSetupTeardownIndependent() {
+			testTypeCmd.Flags().String(cobraext.ConfigFileFlagName, "", cobraext.ConfigFileFlagDescription)
+			testTypeCmd.Flags().Bool(cobraext.SetupFlagName, false, cobraext.SetupFlagDescription)
+			testTypeCmd.Flags().Bool(cobraext.TearDownFlagName, false, cobraext.TearDownFlagDescription)
+			testTypeCmd.Flags().Bool(cobraext.NoProvisionFlagName, false, cobraext.NoProvisionFlagDescription)
+			testTypeCmd.MarkFlagsMutuallyExclusive(cobraext.SetupFlagName, cobraext.TearDownFlagName, cobraext.NoProvisionFlagName)
+			testTypeCmd.MarkFlagsRequiredTogether(cobraext.ConfigFileFlagName, cobraext.SetupFlagName)
+
+			// config file flag should not be used with tear-down or no-provision flags
+			testTypeCmd.MarkFlagsMutuallyExclusive(cobraext.ConfigFileFlagName, cobraext.TearDownFlagName)
+			testTypeCmd.MarkFlagsMutuallyExclusive(cobraext.ConfigFileFlagName, cobraext.NoProvisionFlagName)
+
+			// variant flag should not be used with tear-down or no-provision flags
+			// cannot be defined here using MarkFlagsMutuallyExclusive as in --config-file
```

Comment:
This comment seems a bit lonely around here, not clear to what it refers to. Or is another mutual exclusion missing here?

Reply:
I wanted to add a note that I could not set the mutually exclusive using `testTypeCmd.MarkFlagsMutuallyExclusive`. Variant flag is defined in the `cmd` variable and not in the `testTypeCmd`. 

This is managed afterwards in the code: 
https://github.com/elastic/elastic-package/pull/1250/files#diff-33701d098205d3537f071235876d903ddbfa851c492917e49852c9212bc1d4fdR316-R320

I'll update this comment with that reference